### PR TITLE
add validations and move to WDL 1.0

### PR DESCRIPTION
**Pull request by DavyCats**

### CODE REVIEW

File: structs.wdl [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+struct Root {
+    Array[Sample] samples
+}
+
+struct GermlineDNAinputs {
+    Reference reference
+    BwaIndex bwaIndex
+    File dbSNP
```
Comment (ffinfo) [inline structs.wdl]: Can this be replaced with this IndexedVcfFile struct?

File: structs.wdl [hunk 2]
```diff
@@ -0,0 +1,34 @@
+version 1.0
+
+import "tasks/bwa.wdl" as bwa
+import "tasks/common.wdl" as common
+
+struct Readgroup {
+    String id
+    File R1
+    String R1_md5
```
Comment (ffinfo) [inline structs.wdl]: I think it's more user friendly to make this option optional, same I did for the RNA pipeline

File: structs.wdl [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+struct Library {
+    String id
+    Array[Readgroup]+ readgroups
+}
+
+struct Sample {
+    String id
+    Array[Library]+ libraries
+    String? control
```
Comment (ffinfo) [inline structs.wdl]: where is this used?

File: readgroup.wdl [hunk 4]
```diff
@@ -1,68 +1,87 @@
...
+        Readgroup readgroup
+        String libraryId
+        String sampleId
+        String readgroupDir
+        Int numberChunks = 1
+        GermlineDNAinputs germlineDNAinputs
+    }
 
-    call biopet.SampleConfig as config {
+    call common.CheckFileMD5 as md5CheckR1 {
```
Comment (ffinfo) [inline readgroup.wdl]: This should be a optional step, see also comment on structs.wdl

File: readgroup.wdl [hunk 5]
```diff
@@ -1,68 +1,87 @@
...
-    String sampleId
-    String outputDir
-    Int? numberChunks
+workflow Readgroup {
+    input {
+        Readgroup readgroup
+        String libraryId
+        String sampleId
+        String readgroupDir
+        Int numberChunks = 1
```
Comment (ffinfo) [inline readgroup.wdl]: just to be sure, can the user still set this?

File: pipeline.wdl [hunk 6]
```diff
@@ -1,52 +1,76 @@
...
-                sampleId = sm,
-                refFasta = refFasta,
-                refDict = refDict,
-                refFastaIndex = refFastaIndex,
-                dbsnpVCF = dbsnpVCF,
-                dbsnpVCFindex = dbsnpVCFindex
+                sampleDir = outputDir + "/samples/" + sm.id,
+                sample = sm,
+                germlineDNAinputs = germlineDNAinputs
+                #refFasta = germlineDNAinputs.reference.fasta,
```
Comment (ffinfo) [inline pipeline.wdl]: Please remove out commented code

File: readgroup.wdl (hunk 5 above)
Comment (DavyCats) [reply readgroup.wdl]: yes, this can still be set by the user

**Pull request merged**